app/src/main/java/com/pps/globant/fittracker/mvp/presenter/LoginPresenter.java

```diff
@@ -0,0 +1,131 @@
+        // profile. ID and basic profile are included in DEFAULT_SIGN_IN.
+        gso = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestIdToken(googleServiceClientId)
+                .requestEmail()
+                .build();
+
+        // Build a GoogleSignInClient with the options specified by gso.
+        mGoogleSignInClient = GoogleSignIn.getClient(view.getActivity(), gso);
+    }
+
+    @Subscribe
+    public void onGoogleSignInButtonPressed(GoogleSignInButtonPressedEvent event) {
+        //Starting the intent prompts the user to select a Google account to sign in with.
+        //If you request other stuff beyond profile, email, token and openid,
+        //the user is also prompted to grant access to the requested resources.
+        Intent signInIntent = mGoogleSignInClient.getSignInIntent();
+        view.getActivity().startActivityForResult(signInIntent, RC_GET_TOKEN);
+    }
+
+    @Subscribe
+    public void onGoogleSignOutButtonPressed(GoogleSignOutButtonPressedEvent event) {
+        mGoogleSignInClient.signOut()
+                .addOnCompleteListener(view.getActivity(), new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        Toast.makeText(view.getActivity(), "Signed out from google", Toast.LENGTH_SHORT).show();
+                    }
+                });
+        model.signOutGoogle();
+        view.hideGoogleSignOutButton();
+        view.showGoogleSignInButton();
+        view.setStatusLabel("Signed out");
+        view.setDetailLabel("");
+    }
+
+    @Subscribe
+    public void onSuccessfulGoogleSignIn(SuccessfulGoogleSignInEvent event) {
+        //Once model posts that the login was successful, it sets the detail label with the name.
+        view.setDetailLabel(model.getMail());
+        view.hideGoogleSignInButton();
+        view.showGoogleSignOutButton();
+        view.setStatusLabel("Signed in");
+    }
+
+    public void setActivityResults(int requestCode, int resultCode, Intent data) {
+        //Once the activity started in onGoogleSignInButtonPressed, model.signInGoogle is called.
+        //This method is called by MainActivity, that is listening for the activity to get the result.
+        if (requestCode == RC_GET_TOKEN) {
+            model.signInGoogle(data);
+        }
+    }
+
+    @Subscribe
+    public void onFetchingFbUserDataCompletedEvent(FacebookLoginProvider.FetchingFbUserDataCompletedEvent event) {
+        Resources res = view.getActivity().getResources();
+        view.setLabelFb(String.format(res.getString(R.string.loged_in_name), event.fbUser.getName()));
+    }
+
+    @Subscribe
+    public void onCantRetrieveAllTheFieldsRequestedsEvent(FacebookLoginProvider.CantRetrieveAllTheFieldsRequestedsEvent event) {
+        Resources res = view.getActivity().getResources();
+        view.setLabelFb(res.getString(R.string.fb_login_error_message));
```

Comment:
You could create a new setLabelFb() into the view with an int parameter and you could pass directly  the resource string.
            view.setLabelFb(R.string.fb_login_error_message);